honeybadgermpc/mpc.py

```diff
@@ -64,11 +64,27 @@ def __init__(self, sid, n, t, myid, pid, send, recv, prog, config, **prog_args):
         self.GFElementFuture = type(
             'GFElementFuture', (GFElementFuture,), {'context': self})
 
+    def _get_share_id(self):
+        """Returns a monotonically increasing int value
+        each time this is called
+        """
+        self._share_id += 1
+        return self._share_id
+
     def call_mixin(self, name, *args, **kwargs):
+        """Convenience method to check if a mixin is present, and call it if so
+        args:
+            name(str): Name of the mixin to call
+            args(list): arguments to pass to the call to the mixin
+            kwargs(dict): named arguments to pass to the call to the mixin
+
+        outputs:
+            future that resolves to the result of calling the mixin operation
+        """
         if name not in self.config:
             raise NotImplementedError(f"Mixin {name} not present!")
 
-        return asyncio.ensure_future(self.config[name](self, *args, **kwargs))
+        return asyncio.create_task(self.config[name](self, *args, **kwargs))
```

Comment:
why one over the other?

Reply:
The asyncio documentation specifies that [`create_task`](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task) is preferred over `ensure_future` in python3.7. Furthermore, we mostly already mostly used `create_task`, so this makes it consistent throughout. 

When we call `ensure_future` with a coroutine,[ it simply calls `create_task` anyways](https://github.com/python/cpython/blob/56065d4c8ac03042cb7e29ffda9b1ac544a37b4d/Lib/asyncio/tasks.py#L616).